pkg/prometheus/server/operator.go

```diff
@@ -92,6 +92,8 @@ type Operator struct {
 	canReadStorageClass           bool
 	disableUnmanagedConfiguration bool
 
+	prometheusTopologyShardingFeatureGateEnabled bool
```

Comment:
Instead of adding a new field for each feature gate, we could extend `prompkg.Config` to hold a copy of Gates. When the controller needs to check if a gate is enabled, it would call `c.Config.Gates.Enabled(...)`.
It would also have the nice benefit that the operator always reconciles the workload resources when gates are enabled/disabled.

Reply:
Thanks @simonpasquier 
Got it
Remove flag done.